lib/liquid/template.rb

```diff
@@ -144,6 +144,11 @@ def render!(*args)
     def tokenize(source)
       source = source.source if source.respond_to?(:source)
       return [] if source.to_s.empty?
+
+      if source =~ WhiteSpaceTrimCheck
+          source = source.gsub(WhiteSpaceTrimTags, WhiteSpaceTrimReplace)
```

Comment:
I'll added some tests for this. This really means the logic needs to be in the parser.